h2o-core/src/main/java/water/H2O.java

```diff
@@ -1217,16 +1215,42 @@ private PrioritizedForkJoinPool(int p, int cap) {
     return task;
   }
 
-  public static abstract class H2OFuture<T> implements Future<T> {
-    public final T getResult(){
-      try {
-        return get();
-      } catch (InterruptedException | ExecutionException e) {
-        throw new RuntimeException(e);
-      }
+  public static <T extends RemoteRunnable> T runOnH2ONode(T runnable) {
+    H2ONode node = H2O.ARGS.client ? H2O.CLOUD.leader() : H2O.SELF;
+    return runOnH2ONode(node, runnable);
+  }
+
+  // package-private for unit tests
+  static <T extends RemoteRunnable> T runOnH2ONode(H2ONode node, T runnable) {
+    if (node == H2O.SELF) {
+      // run directly
+      runnable.run();
+      return runnable;
+    } else {
+      RunnableWrapperTask<T> task = new RunnableWrapperTask<>(runnable);
+      H2ONode leader = H2O.CLOUD.leader();
```

Comment:
Small note: Should we use the node from the argument?

Reply:
Good catch, yes, that was the intention. Thanks for catching that.